snuba/subscriptions/dataset_subscription.py

```diff
@@ -0,0 +1,98 @@
+from typing import Any, List, Mapping, Optional, Type, Union
+
+from snuba.query.conditions import ConditionFunctions, binary_condition
+from snuba.query.exceptions import InvalidQueryException
+from snuba.query.expressions import Column, FunctionCall, Literal
+
+
+class DatasetSubscription:
+    def __init__(self, subscription_type: str, data_dict: Mapping[str, Any]) -> None:
```

Comment:
Ideally we don't support legacy at all, but if it's unavoidable we should probably use the enum here rather than a string

Reply:
I don't think we can at the moment.. as far as i know some on prem customers still use it, and so turning off the validation for legacy would permit customers creating faulty subscriptions